src/zenml/zen_server/auth.py

```diff
@@ -761,6 +769,18 @@ def authenticate_external_user(external_access_token: str) -> AuthContext:
             )
             context.alias(user_id=external_user.id, previous_id=user.id)
 
+    # This is the best spot to update the onboarding state to mark the
+    # "zenml login" step as completed for ZenML Pro servers, because the
+    # user has just successfully logged in. However, we need to differentiate
+    # between web clients (i.e. the dashboard) and CLI clients (i.e. the
+    # zenml CLI).
+    user_agent = request.headers.get("User-Agent", "").lower()
+    logger.info(f"User agent: {user_agent}")
```

Comment:
Should this be a debug log instead?

Reply:
It was merely a left-over debug log. Removed now.